improve: rework on items to extinquish OTB file and use client 12 protobuf instead.

marcosvf132 opened the pull request:
# Description

We are using the OTB system for a really long time on OT community, but since Canary is heading to another path like using one single id (ClientID) for map/server/client, i'm propousing to follow Tibia updates more closely.

 Just like cip did on the client (Same is happening on OTC), i think we can forget about using the jurassic Tibia.dat, Tibia.spr and items.otb and use [Google Protobuf](https://github.com/protocolbuffers/protobuf) to deal with items updates without the need of any third partner tools like item editor and client converter.

 Using Tibia protobuf files instead of our old .otb file looks like to me the best approach to keep the repository most updated as possible since, with this changes, we can just simple copy the **_'appearances-XXXXXXXXX.dat'_** file from the client's /assets/ folder, rename it to **_'appearances.dat'_** and paste on ``` /data/items/ ``` on the server and then boom, it's already up-to-date.

Also, the PR include some changes on source to clean the unnused/useless tags and functions that was used on the old serverID/clientID system but are now deprecated.

 > This PR changes does not include client protocol messages updates, only items and a few more data.

# Benefits:
- Fast and easy updates.
- Auto-insert items without the need to register on XML file and with some basic data like: **name**, **description**, **slot position**, **loot type**, **item tier upgrade**, **podium** and all data from old **OTB** file.
- Opens the possibilities to (finally) insert new features like party hunt analyzer, adquire global position for NPCs on the global map with all item prices included, etc.
- Prevent errors and crashes related to wrong **lookType**, **const_me** and **const_ani** effects.

# Tests
Tested and working with no problem so far. But keep in mind that the OTBR Datapack is not ready for this PR yet. If the idea is approved (the idea, not the PR), i will open a similar PR on OTBR datapack to make it compatible with the changes here.

# Map changes
 The map changes was necessary due to some invalid (garbage) items on the map. Does not affect anything at all since they were invisibly, untouchable and unnecessary.

# Protofile
Protofile was added on ``` /src/io/protobuf/protofile/ ``` directory.

## Compiling

# Windows (vcpkg)
> [x86] ./vcpkg install protobuf:x86-windows protobuf[zlib]:x86-windows --recurse
> [x64] ./vcpkg install protobuf:x64-windows protobuf[zlib]:x64-windows --recurse

# Ubuntu/Debian (apt)
> apt-get install libprotobuf-dev protobuf-compiler

# Issues and chat
 Any problem that you have on compiling or have anything you want to change on it you can use the discord. But if it's a hint or some tips to improve the changes or just a discussion please consider using the chat here!


# How to compile protofile
1. Download the [Google Protobuf](https://github.com/protocolbuffers/protobuf) release version protoc-3.18.0 or the version from your protobuf lib. 
2. Place the appearances.proto file on the /bin/ folder from the Google protobuf release.
3. Run cmd  ``` ./protoc.exe --cpp_out=. appearances.dat ```
4. Place it on /src/io/protobuf/ folder.
5. Recompile the server

# Vcpkg protobuf version
![image](https://user-images.githubusercontent.com/66353315/157991358-143358fb-1fb4-48b8-8bab-f61432203bf7.png)

marcosvf132 committed:
Initial commit

marcosvf132 committed:
Simplification of a few lines and changing loottype/slottype declaration.

marcosvf132 committed:
Expanding the protobuf changes

marcosvf132 committed:
Reverting unwanted change

marcosvf132 committed:
:)

marcosvf132 committed:
Fixess and missing check on outfits

marcosvf132 committed:
Merge branch 'master' into Assets-protobuf

marcosvf132 committed:
We dont need to check on every effect if its safe or not.

marcosvf132 committed:
Merge branch 'Assets-protobuf' of https://github.com/marcosvf132/canary into Assets-protobuf

marcosvf132 committed:
Fix rebase changes

marcosvf132 committed:
Build fix

dudantas committed:
Fix sonar check "use the init-statement to declare "it" inside the if statement"

marcosvf132 committed:
Merge branch 'master' into Assets-protobuf

dudantas committed:
Set from "getClientID" to "getID"

dudantas committed:
Fix sonar checks

dudantas committed:
Test fix sonar

dudantas committed:
Fix sonar check

dudantas committed:
Fix sonar checks

dudantas committed:
Indent to tab

dudantas committed:
Add endline

marcosvf132 committed:
Merge branch 'Assets-protobuf' of https://github.com/marcosvf132/canary into Assets-protobuf

marcosvf132 committed:
FIx server load

marcosvf132 committed:
Fix ring equip issue

marcosvf132 committed:
Remove 'corpseType' tag and change .dat folder to /items/

dudantas committed:
Auto compiler appearances.proto and copy for the "src/protobuf" folder

dudantas committed:
Set installRoot path

Costallat reviewed (approved):
Autocompiler

dudantas committed:
Add for compile only "appearances.proto"

dudantas committed:
Merge branch 'master' into pr/204

guispiller commented:
Server is crashing on any combat damage.
Many ways to reproduce:
- Throw a firebomb rune and take 1 damage;
- Start attacking any monster

Crash logs: 
![crash 1](https://user-images.githubusercontent.com/11639062/160431157-4257ff04-c4aa-4638-b80a-5e180bbd29e2.PNG)
![crash 2](https://user-images.githubusercontent.com/11639062/160431165-1b04031b-faf8-4ff8-b921-8d157deee275.PNG)

guispiller commented:
Another error on distro, when you use any health potion:

[2022-28-03 12:41:09.669] [error] Lua script error:
scriptInterface: [Scripts Interface]
scriptId: [D:\Ferramentas OT\Canary\data\scripts\creaturescripts\quests\cults_of_tibia\zarcorix_reflect_damage.lua:callback]
timerEvent: []
 callbackId:[]
function: []
error [attempt to call a number value
stack traceback:
        [C]: at 0x7ff767c368d0
        [C]: in function 'doTargetCombatHealth'
        ...amentas OT\Canary\data\scripts\actions\other\potions.lua:251: in function <...amentas OT\Canary\data\scripts\actions\other\potions.lua:224>]

marcosvf132 committed:
Merge branch 'master' into Assets-protobuf

marcosvf132 committed:
Merge branch 'master' into Assets-protobuf

marcosvf132 commented:
> Another error on distro, when you use any health potion:
> 
> [2022-28-03 12:41:09.669] [error] Lua script error: scriptInterface: [Scripts Interface] scriptId: [D:\Ferramentas OT\Canary\data\scripts\creaturescripts\quests\cults_of_tibia\zarcorix_reflect_damage.lua:callback] timerEvent: [] callbackId:[] function: [] error [attempt to call a number value stack traceback: [C]: at 0x7ff767c368d0 [C]: in function 'doTargetCombatHealth' ...amentas OT\Canary\data\scripts\actions\other\potions.lua:251: in function <...amentas OT\Canary\data\scripts\actions\other\potions.lua:224>]

The crash when using fire bomb rune and the error on log when using potion was due to the fact that my branch was outdated. This issue was fixed on the last commit and the branch is ready to be tested again.

marcosvf132 committed:
Moving QUIVER system from xml to dat

marcosvf132 committed:
Add lua env to ItemType:isQuiver()

Used on OTBR Datapack

marcosvf132 committed:
Merge branch 'opentibiabr:main' into Assets-protobuf

marcosvf132 committed:
Merge branch 'main' into Assets-protobuf

marcosvf132 committed:
Merge branch 'master' into Assets-protobuf

marcosvf132 committed:
Fix missing g_game decouple

marcosvf132 committed:
Map loading errors

- Change the map loader to warn and ignore errors when unserealizing itemIDs from .otbm.

marcosvf132 committed:
Update to 12.86

marcosvf132 committed:
Add new outfits and mounts

dudantas committed:
fix

dudantas committed:
Somes config fixes

dudantas committed:
Fix typo

dudantas committed:
Merge branch 'main' into pr/204
